clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -828,6 +828,18 @@ static void instantiateSYCLAddIRAttributesGlobalVariableAttr(
   S.AddSYCLAddIRAttributesGlobalVariableAttr(New, *A, Args);
 }
 
+static void instantiateSYCLAddIRAnnotationsMemberAttr(
+    Sema &S, const MultiLevelTemplateArgumentList &TemplateArgs,
+    const SYCLAddIRAnnotationsMemberAttr *A, Decl *New) {
+  EnterExpressionEvaluationContext Unevaluated(
```

Comment:
Just wondering, why the variable is called `Unevaluated` when the context is actually constant evaluated?

Reply:
I suspect it was a propagated copy-paste mistake. I have renamed it, but a lot of the surrounding attributes have the same so maybe we want a quick clean-up patch.

Reply:
>>I suspect it was a propagated copy-paste mistake. I have renamed it, but a lot of the surrounding attributes have the same so maybe we want a quick clean-up patch.

Agreed. I have created https://github.com/intel/llvm/issues/5893 to clean up them.

Reply:
I think this clean up also should be done in upstream. I've met this  before.